mava/adders/reverb/sequence.py

```diff
@@ -65,183 +67,83 @@ def __init__(
             sequence_length, sequences are exactly non-overlapping.
           delta_encoded: If `True` (False by default) enables delta encoding, see
             `Client` for more information.
-          chunk_length: Number of timesteps grouped together before delta encoding
-            and compression. See `Client` for more information.
           priority_fns: See docstring for BaseAdder.
+          max_in_flight_items: The maximum number of items allowed to be "in flight"
+            at the same time. See `block_until_num_items` in
+            `reverb.TrajectoryWriter.flush` for more info.
+          end_of_episode_behavior:  Determines how sequences at the end of the
+            episode are handled (default `EndOfEpisodeBehavior.ZERO_PAD`). See
+            the docstring for `EndOfEpisodeBehavior` for more information.
+          chunk_length: Deprecated and unused.
           pad_end_of_episode: If True (default) then upon end of episode the current
             sequence will be padded (with observations, actions, etc... whose values
             are 0) until its length is `sequence_length`. If False then the last
             sequence in the episode may have length less than `sequence_length`.
           break_end_of_episode: If 'False' (True by default) does not break
             sequences on env reset. In this case 'pad_end_of_episode' is not used.
-          max_in_flight_items: The maximum number of items allowed to be "in flight"
-            at the same time. See `reverb.Writer.writer` for more info.
+          use_next_extras: If true extras will be processed the same way observations
+          are processed. If false extras will be processed as actions are processed.
         """
-
-        if delta_encoded:
-            NotImplementedError(
-                "Note (dries): Delta encoding has not been verified to "
-                "work in Mava yet. If you want to use delta encoding"
-                " first verify that it is working and then contribute "
-                "the update with a working example of delta encoding"
-                " to the Mava repo."
-            )
-
-        super().__init__(
+        ReverbParallelAdder.__init__(
+            self,
             client=client,
-            buffer_size=sequence_length,
-            max_sequence_length=sequence_length,
+            # We need an additional space in the buffer for the partial step the
+            # base.ReverbAdder will add with the next observation.
+            max_sequence_length=sequence_length + 1,
             delta_encoded=delta_encoded,
-            chunk_length=chunk_length,
             priority_fns=priority_fns,
-            max_in_flight_items=max_in_flight_items,
+            max_in_flight_items=max_in_flight_items,  # type: ignore
```

Comment:
Why is there a type mismatch here? Would it not always be an `int`?

Reply:
It is a mismatch between `Optional[int]` and `int`. Sorted it out in new commit though.